### DISCUSSION TITLE
who knows how to connect inhand sprites in meta.json and yml?

### DISCUSSION THREAD

**Discussion by onesch** [space-wizards/space-station-14#33601] [Questions] [answered]
I tried, but it didn't work, I want to hook inhand sprites for [floodlight_broken](https://github.com/space-wizards/space-station-14/blob/31d5a668669cf14558e25c6607a446aac0927b5d/Resources/Textures/Objects/Misc/Lights/lights.rsi/floodlight_broken.png
) and [floodlight_burned](https://github.com/space-wizards/space-station-14/blob/31d5a668669cf14558e25c6607a446aac0927b5d/Resources/Textures/Objects/Misc/Lights/lights.rsi/floodlight_burned.png).
for the off and on states everything is clear, but what to do with a broken and burnt one? what should I write in map?
![image](https://github.com/user-attachments/assets/f0a3d870-27ab-409e-a745-001d9351fb2b)

[Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml](https://github.com/space-wizards/space-station-14/blob/31d5a668669cf14558e25c6607a446aac0927b5d/Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml)
[Resources/Textures/Objects/Misc/Lights/lights.rsi](https://github.com/space-wizards/space-station-14/tree/31d5a668669cf14558e25c6607a446aac0927b5d/Resources/Textures/Objects/Misc/Lights/lights.rsi)

**Comment by MilenVolf** [comment, marked as answer]
Currently burnt sprite is not used anywhere
We have FloodlightBroken that located in the same file, just a little bit below
[Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml#L243](https://github.com/space-wizards/space-station-14/blob/31d5a668669cf14558e25c6607a446aac0927b5d/Resources/Prototypes/Entities/Objects/Misc/fluff_lights.yml#L243)

If you want to use inhand sprites for it, just use something like this:
```yaml
  - type: Item
    sprite: Objects/Misc/Lights/lights.rsi # Here we locating the folder with inhand sprites
```

If you need to use custom prefix (for example: we have `floodlight-broken-inhand-left.png` and  `floodlight-broken-inhand-right.png`) , you can do something like this:
```yaml
  - type: Item
    sprite: Objects/Misc/Lights/lights.rsi # Here we locating the folder with inhand sprites
    heldPrefix: floodlight-broken # Here we specifying the prefix of the name of our inhand sprites png files
```

Hope it will help you a little!

**Reply by onesch** [reply]
I added sprites and added lines in the `yml` file but they did not appear
![image](https://github.com/user-attachments/assets/8744eab7-edd8-42bf-ac8f-0fd7429b95ad)
![image](https://github.com/user-attachments/assets/313d152e-714d-45ec-95be-3295f2e590f1)


Maybe I need to add something else to `meta.json`? So I added it but it still doesn’t show
![image](https://github.com/user-attachments/assets/80ef2186-40ca-41d6-98b8-3639dbb0e846)

and even after I added lines `floodlight-broken-inhand-left.png` and `floodlight-broken-inhand-right.png` to the `meta.json` it stopped displaying altogether
![image](https://github.com/user-attachments/assets/e3a96bb7-b0fd-4e32-bba2-4607812dcdcb)

**Reply by muburu** [reply]
if even one thing is wrong in the meta file it won't work AT ALL
the right inhand says "rght" in the meta file instead of right, if it can't reference something that it can't find it won't work

**Reply by MilenVolf** [reply]
> I added sprites and added lines in the yml file but they did not appear ![image](https://private-user-images.githubusercontent.com/118821520/390561421-8744eab7-edd8-42bf-ac8f-0fd7429b95ad.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI3NDA5MDgsIm5iZiI6MTczMjc0MDYwOCwicGF0aCI6Ii8xMTg4MjE1MjAvMzkwNTYxNDIxLTg3NDRlYWI3LWVkZDgtNDJiZi1hYzhmLTBmZDc0MjliOTVhZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyN1QyMDUwMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNmE0NzllOTFkMjNkMjNlOWQ1M2EzMDllMjY5YzlhMGY0OGQwM2U2NzQzODBhNDc5YTQwY2I5ZGYxMDIzMjJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.oabEEiYzcMmsVFTVOP7g-8ATOnrmPM64Fq7ROJDR5DM)

What the console says if you try to spawn it?

**Reply by MilenVolf** [reply]
> if even one thing is wrong in the meta file it won't work AT ALL the right inhand says "rght" in the meta file instead of right, if it can't reference something that it can't find it won't work

Yeah, thanks for noticing

**Reply by onesch** [reply]
the problem was in the unfinished `i` lol, 
so, what to do with the burnt one? I will have to wait a while for it to burn first and then..
I haven’t seen anywhere that after a while `floodlight` becomes `floodlight_burned` probably won't do it.
![image](https://github.com/user-attachments/assets/08dcd5d2-75ee-49b6-a7a9-12f2beb0b3aa)
